### DISCUSSION TITLE
Type error: Argument of type 'URL' is not assignable to parameter of type 'RequestInfo'.

### DISCUSSION THREAD

**Discussion by alsonick** [vercel/next.js#44719] [Help] [answered]
I'm using Vercel's OG Image Generation feature to dynamically generate social cards for my site. I'm using a custom font and I'm following one of the [examples](https://vercel.com/docs/concepts/functions/edge-functions/og-image-examples#using-a-custom-font) provided by Vercel. 

Whenever I build my project I get the following build error:

`Type error: Argument of type 'URL' is not assignable to parameter of type 'RequestInfo'.`

Helpful screenshots:

(The code causing the error - Screenshot 1)
<img width="464" alt="Screenshot 2023-01-09 at 11 13 32" src="https://user-images.githubusercontent.com/101022772/211295650-17fd7c7e-8770-43cf-a122-7fb85052b73b.png">

(Screenshot of the error log - Screenshot 2)
<img width="645" alt="Screenshot 2023-01-09 at 11 14 04" src="https://user-images.githubusercontent.com/101022772/211295727-dc0cfccd-f505-4178-b249-d7592993f70d.png">

Desired behaviour: Successful builds

**Comment by icyJoseph** [comment, marked as answer]
You need to call `.toString()` on `new URL('something/something')`

## Edit

It is worth noting that the fetch in the browser, does take URL objects as arguments. IIRC, the node-fetch polyfill doesn't, _or something like that_, more info here: https://github.com/node-fetch/node-fetch/issues/1261#issuecomment-914648515

As they say there, `it is a TS thing` ~ If you try with Node 18, which implements fetch natively:

```js
const pokeAPI = "https://pokeapi.co";

fetch(new URL("/api/v2/pokemon/ditto", pokeAPI));
```

It'll work - but if you make the file into TS, TS complains.

**Reply by alsonick** [reply]
Tysm!

**Reply by icyJoseph** [reply]
Don't forget to mark as answered! Have a good day!
